internal/app/cache/cache.go

```diff
@@ -0,0 +1,31 @@
+package cache
+
+import (
+	envoy_service_discovery_v3 "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+)
+
+type Cache interface {
```

Comment:
1) Added error to the return values for several of these methods. It may not be necessary, but while I was working with freecache I found that we might need to propagate errors corresponding to being unable to decode the cache value, encode a cache value, the key not existing, etc.

2) Not sure how to add this to a go interface but the cache should be instantiated with TTL and memory as configurable values.

Reply:
1. Makes sense
2. I don't believe these need to be a part of the interface. They can be implementation details in the internal struct implementation when constructing the Cache object.